### DISCUSSION TITLE
Action when cue is detected

### DISCUSSION THREAD

**Discussion by pcb105** [futzu/threefive#92] [Q&A] [answered]
I want to perform an action whenever a cue is detected from a UDP stream.  I wrote the following code....

import sys
import threefive

if __name__ == "__main__":
    if len(sys.argv) > 1:
        for arg in sys.argv[1:]:
        	strm = threefive.Stream(arg)
        	SCTE35 = strm.decode(func=print("cue"))
    else:
        threefive.decode(sys.stdin.buffer)

python3 NodeSCTE35.py udp://:1234

When I run the code.....
Output: 
cue

{It then exits when an actual cue happens}

When I run the code without the func=print("cue")
I get the SCTE information printed out each time the UDP stream has a SCTE marker in it.  

I would like to perform an action every time I get a SCTE message within the UDP stream.  

Any help would be appreciated.  

-Paul

**Comment by pcb105** [comment, marked as answer]
Sorry I figured it out here is code that does what I was asking for.... If anyone cares :) 

import json
import sys
import threefive

def add_cue(cue):
    print(cue.get())
    
if __name__ == "__main__":
    if len(sys.argv) > 1:
        for arg in sys.argv[1:]:
        	strm = threefive.Stream(arg)
        	strm.decode(func=add_cue)
    else:
        threefive.decode(sys.stdin.buffer)

python3 NodeSCTE35.py udp://:1234

**Reply by futzu** [reply]
Hey, I'm sorry I didn't see this until now, however that is exactly how you do it. Nice job man. 

* You can also do it with a call to decode_next()

```lua
import sys
import threefive

def do():
    arg = sys.argv[1]
    with open(arg,'rb',encoding="utf-8") as tsdata:
        st = threefive.Stream(tsdata)
        while True:
            cue = st.decode_next()
            if not cue:
                return False
            if cue:
                print(cue.get())

if __name__ == "__main__":
    do()

```
